lib/matplotlib/cm.py

```diff
@@ -258,9 +261,14 @@ def to_rgba(self, x, alpha=None, bytes=False):
 
         # This is the normal case, mapping a scalar array:
         x = ma.asarray(x)
-        x = self.norm(x)
-        x = self.cmap(x, alpha=alpha, bytes=bytes)
-        return x
+        if norm:
+            x = self.norm(x)
+        rgba = self.cmap(x, alpha=alpha, bytes=bytes)
+        # For floating-point greyscale images, we treat negative as
+        # transparent so we copy that over to the alpha channel
+        if x.ndim == 2 and x.dtype.kind == 'f':
+            rgba[:, :, 3] = np.where(x < 0.0, 0, rgba[:, :, 3])
```

Comment:
Thanks -- I'll change that.  Those crazy indexing tricks in Numpy always evade me.